src/plugins/Polls/UI/PollsDialog.tsx

```diff
@@ -0,0 +1,348 @@
+            '& >div': {
+                width: '80%',
+                margin: theme.spacing(2),
+            },
+        },
+        addButton: {
+            position: 'absolute',
+            bottom: '0',
+            right: '10px',
+        },
+        loading: {
+            color: '#fff',
+        },
+    }),
+)
+
+interface NewPollProps {
+    loading: [boolean, React.Dispatch<React.SetStateAction<boolean>>]
+    senderName: string | undefined
+    senderId: string | undefined
+    createNewPoll: () => void
+}
+
+function NewPollUI(props: PollsDialogProps & NewPollProps) {
+    const classes = useStylesExtends(useNewPollStyles(), props)
+    const [loading, setLoading] = props.loading
+    const [question, setQuestion] = useState('')
+
+    const [optionsInput, setOptionsInput] = useState<Object>({ 0: '', 1: '' })
+    const options = Object.values(optionsInput)
+
+    const [days, setDays] = useState(1)
+    const [hours, setHours] = useState(0)
+    const [minutes, setMinutes] = useState(0)
+
+    const handleOptionsInput = (index: number, e: any) => {
+        setOptionsInput({
+            ...optionsInput,
+            [index]: (e.target as HTMLInputElement)?.value,
+        })
+    }
+
+    const addNewOption = () => {
+        const length = options.length
+        setOptionsInput({
+            ...optionsInput,
+            [length]: '',
+        })
+    }
+
+    const sendPoll = async () => {
+        const start_time = new Date()
+        const end_time = new Date(
+            start_time.getTime() + days * 24 * 60 * 60 * 1000 + hours * 60 * 60 * 1000 + minutes * 60 * 1000,
+        )
+        setLoading(true)
+        Services.Plugin.invokePlugin('maskbook.polls', 'createNewPoll', {
+            question,
+            options: optionsInput,
+            start_time,
+            end_time,
+            sender: props.senderName,
+            id: props.senderId,
+        }).then((res) => {
+            setLoading(false)
+            props.createNewPoll()
+        })
+    }
+
+    const renderSelect = (count: number, fn: (newVal: number) => void, defaultIndex = 0) => {
+        const options = new Array(count).fill('')
+
+        return (
+            <Select
+                MenuProps={{ container: props.DialogProps?.container ?? PortalShadowRoot }}
+                value={defaultIndex}
+                onChange={(e) => fn(e.target.value as number)}>
+                {options.map((item, index) => (
+                    <MenuItem value={index} key={index}>
+                        {index}
+                    </MenuItem>
+                ))}
+            </Select>
+        )
+    }
+
+    return (
+        <>
+            <FormControl className={classes.line}>
+                <TextField
+                    label="Ask a question..."
+                    variant="filled"
+                    InputProps={{
+                        onChange: (e) => {
+                            setQuestion((e.target as HTMLInputElement)?.value)
```

Comment:
Why do you use it like this? I think there is a `onChange` event directly on the `<TextField />` component and you don't need this.